### ISSUE TITLE
Possible explanation of Git configuration mystery

### ISSUE THREAD

**Issue by jennybc** [STAT545-UBC/Discussion#239]
In class today, at least 3 people -- all on Macs! -- had trouble putting their new package under version control via `devtools::use_git()`. It probably looked something like this:

``` r
library(devtools)
use_git()
#> * Initialising repo
#> * Adding .Rproj.user, .Rhistory, and .RData to .gitignore
#> * Adding files and committing
#> Error in default_signature(repo) : 
#>   Error in 'git2r_signature_default': Config value 'user.name' was not found
```

My current best guess is that the Mac thing is a coincidence and all 3 had deficient user-level Git configuration and today is the first time we exposed that problem.

When we [introduced ourselves to Git](http://stat545-ubc.github.io/git04_introduce-self-to-git.html), the instructions are like so:

``` sh
git config --global user.name 'Jennifer Bryan'
git config --global user.email 'jenny@stat.ubc.ca'
```

The `--global` part is important and ensures that the user name and email are written in to user-level `.gitconfig`. If you don't specify global, it is possible to set these things locally, i.e. just for a specific Git repo. Which can be useful in some circumstances! But in our case, it's not what we want.

To check that you have user name and email set globally for yourself, do this:

``` sh
git config --global --list
```

and hunt for user name and email. If absent, use the command above to set correctly.

**Comment by jennybc** [comment]
The alternative theory is that something we did ?with `devtools`? some how clobbered the globally configured user name and email. But I was not able to reproduce that on one of the Macs in question or my own. I also can't see anyway for that to happen. But please comment here if people have recurring problems with missing Git config info. I'd want to know about that.

**Comment by Poush87** [comment]
I don't think my issue with using git this morning had to do with setting user name and email globally. I was getting a message saying 'Access denied' as RStudio/Git/Github (I'm not sure which) wasn't recognising the passphrase for the SSH key associated with Github. I generated a new key and it seems to work fine!

**Comment by jennybc** [comment]
@poush87

But we didn't do anything (yet) with GitHub and these new packages we're making with `devtools`. `use_git()` shouldn't touch anything related to a remote. But I guess it's possible that a faulty SSH configuration is problematic (vs. completed absence), even if we're not trying to, e.g., push?

**Comment by Poush87** [comment]
You're right - It didn't strike me that use_git() has to do with the local git repo and not github!
